coalib/bears/Bear.py

```diff
@@ -273,6 +279,191 @@ def log_message(self, log_message, timestamp=None, **kwargs):
+                elif c == separator and nb_brackets == 0:
+                    final.append(i)
+                if nb_brackets < 0:
+                    raise ValueError('Invalid arguments to --profile-bears')
+            final.append(len(sentence))
+            if nb_brackets > 0:
+                raise ValueError('Invalid arguments to --profile-bears')
+        except ValueError as err:
+            self.err(err.args[0])
+            return
+        return([sentence[i:j].strip(separator) for i, j in zip(final,
+                                                               final[1:])])
+
+    def pstats_config(self, prof, stream, profile_bears):
+        flag = 0
+        ps = pstats.Stats(prof, stream=stream)
+        if len(profile_bears) > 1:
+            for setting in profile_bears[1:]:
+                str_setting = str(setting).lower().strip(' \n\t')
+                if u'(' in str(setting):
+                    args = str(setting).split('(', 1)[1].strip(
+                        ' \t\n)').replace('\'', '').replace(
+                        '"', '').replace(' ', '').split(',')
+                if not str_setting == u'':
+                    try:
+                        str_setting = str_setting.split('(', 1)[0]
+                        if 'args' in locals() and args != [u''] and (
+                                str_setting == u'reverse_order' or
+                                str_setting == u'strip_dirs'):
+                            raise ValueError(
+                                'The pstats method "{}" does'
+                                ' not accept any arguments.'
+                                'Discarding user settings.', str_setting)
+                        if 'args' in locals() and args == [u''] and not (
+                                str_setting == u'reverse_order' or
+                                str_setting == u'strip_dirs'):
+                            raise ValueError(
+                                'The pstats method "{}" requires an argument',
+                                str_setting)
+                        flag = 1
+                        if str_setting == u'reverse_order':
+                            ps.reverse_order()
+                        elif str_setting == u'strip_dirs':
+                            ps.strip_dirs()
+                        elif str_setting == u'add':
+                            ps.add(*args)
+                        elif str_setting == u'dump_stats':
+                            ps.dump_stats(*args)
+                        elif str_setting == u'sort_stats':
+                            ps.sort_stats(*args)
+                        elif str_setting == u'print_stats':
+                            ps.print_stats(*args)
+                        elif str_setting == u'print_callers':
+                            ps.print_callers(*args)
+                        elif str_setting == u'print_callees':
+                            ps.print_callees(*args)
```

Comment:
Now I see that the `print_callers()` and `print_callees()` can also work without an argument, will be fixing that soon

Reply:
currently if not provided with an argument they will generate an error and the profiler will go with the default settings